### ISSUE TITLE
Guidance on Zarr benchmarking

### ISSUE THREAD

**Issue by abarciauskas-bgse** [developmentseed/cloud-optimized-geospatial-formats-guide#30]
I've started to work on guidance for Zarr chunk size and shape. A future version should link to or include guidance on how to benchmark different zarr chunk sizes by making it easy to generate zarr stores with different chunk shapes and then execute tests against a configurable library.

Some prior and current art to review:
- [Benchmarking Zarr and Parquet Data Retrieval using the National Water Model (NWM) in a Cloud-native environment](https://www.azavea.com/blog/2022/09/22/benchmarking-zarr-and-parquet-data-retrieval-using-the-national-water-model-nwm-in-a-cloud-native-environment/) 
- [rabernat/zarr_hdf_benchmarks](https://github.com/rabernat/zarr_hdf_benchmarks)
- (Benchmark Zarr Implementations)[https://hackmd.io/@I9Hj1bLETn6QIva97pA3Hw/By7rlRXd5]

**Comment by wildintellect** [comment]
@abarciauskas-bgse @maxrjones we have some material that can be referenced now?

**Comment by abarciauskas-bgse** [comment]
I don't think we have general on chunk size and shape for Zarr. I do think we can reuse the chunk shape and size guidance from the cloud-optimized HDF5 section, perhaps by creating a separate page on chunk size and shape and linking to it from both format-specific pages. However that guidance is just general principles, not how to create benchmarks. Do we still want to include this? Are people asking for it?

But I'm glad you brought this up because we should update the Zarr section to include Zarr v3 format specification and features (although I could also see arguing to wait until we're sure it's not going to change much).

It's also relevant to this issue I believe the new sharding and chunk grid features would have an impact on the design of chunk size and shape, if the data producer chooses to use those features.

Interested to hear what @maxrjones thinks.

**Comment by maxrjones** [comment]
> Do we still want to include this? Are people asking for it?

While we have materials and guidance to offer on benchmarking approaches, I agree that this is the important question to answer before devoting time here. I suggest we broaden this issue to "guidance on benchmarking" and wait to see if anyone submits a request for the addition.

Chunking is always important, but not specific to Zarr. I think it'd be worth having a general concepts section that includes chunking/partitioning, sharding, encoding, overviews and link to those concepts from the formats that they apply to rather than building out explanations for each individual file format. We can add additional caveats to the individual file sections, but I would lean towards a guide structure that helps people understand commonalities between file formats.

**Comment by abarciauskas-bgse** [comment]
Noting that general guidance on benchmarking and examples has been requested multiple times.
